### ISSUE TITLE
Why doesn't function New(server ...string) return err?

### ISSUE THREAD

**Issue by ZacharyZYH** [bradfitz/gomemcache#68]
``` go
// New returns a memcache client using the provided server(s)
// with equal weight. If a server is listed multiple times,
// it gets a proportional amount of weight.
func New(server ...string) *Client {
	ss := new(ServerList)
	ss.SetServers(server...)
	return NewFromSelector(ss)
}
```

**Comment by glasser** [comment]
If you pass in a server that doesn't resolve, you'll end up with an empty server list and the first PickServer call will return `memcache: no servers configured or available`.  It would be nice, though, if the specific resolve error didn't get lost so you know which server failed to resolve.

In theory you could fix this without changing the API by saving the error on the Client and having each public API check the error first.

**Comment by bunyk** [comment]
Yes, because of this I got "memcache: no servers configured or available" during Get call, but could have "address 127.0.0.1: missing port in address" during New call which would save me a lot of time.

**Comment by AaronM04** [comment]
I just encountered this problem. One of the servers in the list couldn't be resolved so the server list was empty.

Later, when a memcache operation was performed, the confusing and unhelpful error `memcache: no servers configured or available` was returned.

I suggest updating the godoc comment for `New` to explain this behavior and to recommend calling `SetServers` for catching these errors.
